lib/ansible/modules/cloud/docker/docker_swarm.py

```diff
@@ -373,7 +373,8 @@ def __update_spec(self, spec):
             self.parameters.name = spec['Name']
 
         if (self.parameters.labels is None):
-            self.parameters.labels = spec['Labels']
+            if (spec['Labels'] is not None):
```

Comment:
That will result in the same error. You need to write
```suggestion
            if spec.get('Labels') is not None:
```